Modules/Module_Processing/xASL_init_LoadDataPar.m

```diff
@@ -134,4 +136,61 @@
+%% -----------------------------------------------------------------------------
+function [x] = xASL_initLoadDataPar_PrepareAtlas4ROI(x)
+%Check for x.S.Atlases & x.S.TissueMasking and provide default values or print instructions
+
+bAtlasTissueMatch = true; % Necessary condition
+
+if ~isfield(x, 'S') || (~isfield(x.S,'Atlases') && ~isfield(x.S, 'TissueMasking'))
+	% Default atlases/ROIs & tissue masks if nothing is provided
+	x.S.Atlases = {'TotalGM','DeepWM'}; % Default
+    x.S.TissueMasking = {'GM' 'WM'}; % GM WM, fits with the TotalGM & DeepWM above
+    % Note that this should be in the same order as the atlases/ROIs
+    % A mismatch (e.g. TissueMasking=GM for Atlases=deepWM) would result in an empty ROI, producing a NaN in the .tsv table
+elseif ~isfield(x.S, 'Atlases') && isfield(x.S, 'TissueMasking')
+	% Missing Atlases, but provided TissueMasking - cannot continue
+    warning('Custom tissue-types (x.S.TissueMasking) specified without ROI atlas-selection (x.S.Atlases). Atlases need to be provided. See instructions below:');
+    bAtlasTissueMatch = false;
+elseif isfield(x.S, 'Atlases') && isfield(x.S, 'TissueMasking') && length(x.S.Atlases)~=length(x.S.TissueMasking)
+	% Non matching lengths, cannot continue
+    warning('The same number of ROI atlases as subject-wise tissue-types provided does not match:');
+    fprintf('%s\n', ['S:{Atlases:["' strjoin(x.S.Atlases, '", "') '"]}']);
+    fprintf('%s\n', ['S:{TissueMasking:["' strjoin(x.S.TissueMasking, '", "') '"]}']);
+    bAtlasTissueMatch = false;
+elseif 	isfield(x.S, 'Atlases') && ~isfield(x.S, 'TissueMasking')
+	% TissueMasking not provided, so it has to be extracted from Atlases as previously.
+    warning('ROIs provided in S.Atlases without the tissue-types for these ROIs in S.TissueMasking.');
+    bAtlasTissueMatch = false;
+	textRecommendation = 'S:{TissueMasking:["';
+	% Fill in TissueMasking based on the atlas names and default to GM. The user will see a warning and automatic tissue masks to verify
+	for iAtlas = 1:numel(x.S.Atlases)
+		if iAtlas > 1
+			textRecommendation = [textRecommendation '", "'];
+		end
+		if ~isempty(regexpi(x.S.Atlases{iAtlas}, 'WM')) || ~isempty(regexpi(x.S.Atlases{iAtlas}, 'whitematter'))
+			textRecommendation = [textRecommendation 'WM'];
+		elseif ~isempty(regexpi(x.S.Atlases{iAtlas}, 'WB')) || ~isempty(regexpi(x.S.Atlases{iAtlas}, 'wholebrain'))
+			textRecommendation = [textRecommendation 'WB'];
+		else
+			textRecommendation = [textRecommendation 'GM'];
+		end
+	end
+	textRecommendation = [textRecommendation '"]}'];
+	fprintf('Recommended addition to the dataPar.json to match provided Atlases: %s\n\n', textRecommendation);
+end
+if ~bAtlasTissueMatch
+    fprintf('%s\n', 'When ROI atlases are provided in S.Atlases, their tissue types');
+    fprintf('%s\n', 'need to be provided as well in dataPar.json as S.TissueMasking, with either option ''GM'', ''WM'', ''WB'' (GM+WM).');
+    fprintf('%s\n', 'When not provided, these default to:');
```

Comment:
This should either be

fprintf('%s\n', 'When not provided, these would have defaulted to:');

or not print it at all when `~bAtlasTissueMatch`, because it may confuse the reader in thinking that we default to this (which we don't necessarily when you issue this message)

Reply:
Yes I think it is the combination of the location that you moved this to. My point was that if you have 

~bAtlasTissueMatch

we do NOT default to anything, so this confuses the user.

Reply:
OK. so I removed it - fully up to you. As I say - you wrote that warning code in that exact location ;)